### PR TITLE
Prevents Regal Rats from being possessed pre-round

### LINKED ISSUE #70018: Players can take control of Regal Rats before the round starts
## Testmerges:
#69988
#69240
#69983

## Reproduction:
If a regal rat spawns before roundstart and a observing player clicks on it then they can control it. This lets them interact with the station before anyone else has even spawned.

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

You can no longer observe and assume control of a regal rat during pre-round.

## Why It's Good For The Game

Closes #70018.

## Changelog

:cl:
fix: You can no longer assume control of Regal Rats during preround.
/:cl:

### CODE REVIEW

**Review by Mothblocks — APPROVED** [review]
In a perfect world I think this would be a component or similar, I don't like that this can easily just happen again when someone adds a new mob.

**Comment by Mothblocks** [comment]
Human fly trap monsters have it too, though I don't know if they re-implement it themselves or are just subtyping off of something

File: code/__DEFINES/dcs/signals/signals_mind.dm [hunk 1]
```diff
@@ -6,3 +6,6 @@
 
 /// Called on the mind when an antagonist is being removed, after the antagonist list has updated (datum/antagonist/antagonist)
 #define COMSIG_ANTAGONIST_REMOVED "antagonist_removed"
+
+/// Called when an observer attempts to possess a ghost_role_spawnpoint component-bearing mob
+#define COMSIG_ATTEMPT_POSSESSION "attempt_possession" //Find a better file to put this in
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals/signals_mind.dm:11]: Just make a `signals_ghost_role_spawnpoint.dm` file

File: code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm [hunk 2]
```diff
@@ -132,21 +132,20 @@
...
 		return ..()
 
-	if(!SSticker.HasRoundStarted())
-		return ..()
-
 	var/become_gorilla = tgui_alert(user, "Become a Cargorilla?", "Confirm", list("Yes", "No"))
 	if(become_gorilla != "Yes" || QDELETED(src) || QDELETED(user) || being_polled_for || mind || client)
 		return
 
+	SEND_SIGNAL(src, COMSIG_ATTEMPT_POSSESSION, user)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:148]: The entire point is that the entire proc this is in shouldn't exist and should be handled by the component

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 3]
```diff
@@ -74,17 +75,11 @@
...
 	var/rat_ask = tgui_alert(usr, "Become the Royal Rat?", "Are you sure?", list("Yes", "No"))
 	if(rat_ask != "Yes" || QDELETED(src))
 		return
-	if(key)
-		to_chat(user, span_warning("Someone else already took the rat!"))
-		return
-	key = user.key
-	src.log_message("took control of [name].", LOG_GAME)
+
+	SEND_SIGNAL(src, COMSIG_ATTEMPT_POSSESSION, user)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:82]: The entire point is that the entire proc this is in shouldn't exist and should be handled by the component

File: code/datums/components/ghost_role_spawnpoint.dm [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/datum/component/ghost_role_spawnpoint //Small component, but it helps to consolidate general observer role spawning behavior/checks into one place.
```
Comment (Mothblocks) [inline code/datums/components/ghost_role_spawnpoint.dm]:
```suggestion
/// Write useful documentation here for what this does and how to use it
/datum/component/ghost_role_spawnpoint
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Well if we're going to do the best path then code review is about to get more serious.

You can't have the component be like this. The component should handle everything on its own, and the only thing consumers should have to do is add it, maybe with the text they want to see.

**Comment by Rhials** [comment]
Alright, I figure that I'm just out of my depth on this one. I intend to return and make this a component when I am smarter, but there's not point in holding up the fix until then.

**Comment by Mothblocks** [comment]
Fair enough

**Pull request merged**